pkg/virt-config/configuration.go

```diff
@@ -231,6 +231,24 @@ func defaultClusterConfig(cpuArch string) *v1.KubeVirtConfiguration {
 				Burst: DefaultVirtWebhookClientBurst,
 			}}},
 		},
+		ArchitectureConfiguration: &v1.ArchConfiguration{
+			Amd64: &v1.ArchSpecificConfiguration{
+				OVMFPath:         DefaultARCHOVMFPath,
+				EmulatedMachines: strings.Split(DefaultAMD64EmulatedMachines, ","),
+				MachineType:      DefaultAMD64MachineType,
+			},
+			Arm64: &v1.ArchSpecificConfiguration{
+				OVMFPath:         DefaultAARCH64OVMFPath,
+				EmulatedMachines: strings.Split(DefaultAARCH64EmulatedMachines, ","),
+				MachineType:      DefaultAARCH64MachineType,
+			},
+			Ppc64le: &v1.ArchSpecificConfiguration{
+				OVMFPath:         DefaultARCHOVMFPath,
+				EmulatedMachines: strings.Split(DefaultPPC64LEEmulatedMachines, ","),
+				MachineType:      DefaultPPC64LEMachineType,
+			},
+			DefaultArchitecture: runtime.GOARCH,
```

Comment:
If I set `ArchitectureConfiguration.DefaultArchitecture="Arm64"`, would I get the Arm64 default automatically?  I think the way the code is structured I believe I would get all the x86 defaults.

Reply:
For example, `GetEmulatedMachines` will return the x86 default with the above configuration but the admission webhook will expect to get the ARM64 default.  So either setting only `ArchitectureConfiguration.DefaultArchitecture="Arm64"` can't be allowed for a config or it is allowed and we need handle it.  I prefer the former because it would allow us to run the kubevirt test suite without changes against only ARM64 nodes.
```go
func (c *ClusterConfig) GetEmulatedMachines(arch string) []string {
	return c.GetConfig().EmulatedMachines
	switch arch {
	case "arm64":
		return c.GetConfig().ArchitectureConfiguration.Arm64.EmulatedMachines
	case "ppc64le":
		return c.GetConfig().ArchitectureConfiguration.Ppc64le.EmulatedMachines
	default:
		return c.GetConfig().ArchitectureConfiguration.Amd64.EmulatedMachines
	}
}
```

Reply:
@rthallisey If I understand your question correctly, the way that the default architecture config item operates is in the event that a user has chosen to not explicitly set `ArchitectureConfiguration.DefaultArchitecture`, it will resort to the compiled architecture of the binary (`runtime.GOARCH`) when defining the VM/VMI. This was added in order to avoid a situation where going forward user's had to always explicitly specify an architecture in the spec of their VM. Based on the specified architecture for the VM, the appropriate defaults will be fetched. Does this answer your question?
\

Reply:
If not, I'm not sure I'm understanding fully the situation you're describing.

Reply:
When a VM is going through the admission webhooks, a default architecture is applied if the user has not explicitly specified it on the VM spec (see below screenshot)

![image](https://user-images.githubusercontent.com/16483722/217121325-099a44f7-07f1-4860-a225-2e96b8541b42.png)


This value is either the default set in the config for `ArchitectureConfiguration.DefaultArchitecture`, or it falls back to the compiled architecture of the binary.

Once this field has been set, all subsequent calls for `GetEmulatedMachines`,etc will reference the set field to determine the appropriate defaults.

Reply:
Right I'm saying if `ArchitectureConfiguration.DefaultArchitecture=arm64` and `spec.Architecture` is unset, the vmi will eventually get `spec.Architecture=arm64`.  However, the MachineType and other defaults will be for amd64 because the vmi definition has `vmi.Spec.Architecture` as `nil` during mutation.

Reply:
AhI think I see what you're saying now. I think the solution to this is merely to set the default architecture first in that block above, no? This will ensure that for machine type and others for the VM mutator which relies on the architecture will be appropriately set.

Reply:
The same goes for the VMI mutator:

**vmi-mutator.go**

BEFORE
```
		// Set VMI defaults
		log.Log.Object(newVMI).V(4).Info("Apply defaults")
		mutator.setDefaultMachineType(newVMI)
		mutator.setDefaultResourceRequests(newVMI)
		mutator.setDefaultGuestCPUTopology(newVMI)
		mutator.setDefaultPullPoliciesOnContainerDisks(newVMI)
		mutator.setDefaultArchitecture(newVMI)
```

AFTER
```
		// Set VMI defaults
		log.Log.Object(newVMI).V(4).Info("Apply defaults")
		mutator.setDefaultArchitecture(newVMI)
		mutator.setDefaultMachineType(newVMI)
		mutator.setDefaultResourceRequests(newVMI)
		mutator.setDefaultGuestCPUTopology(newVMI)
		mutator.setDefaultPullPoliciesOnContainerDisks(newVMI)

```

Reply:
Ya I think that would work.  The other option would be to handle empty strings in the setDefault funcs.

Reply:
@daghaian personally I prefer the latter option so we don't have to rely on ordering.

Reply:
Ah ok, let me look at the latter instead @rthallisey

Reply:
@rthallisey Would this align with your idea of handling the empty case?

```golang
func (c *ClusterConfig) GetEmulatedMachines(arch string) []string {
	// if arch is empty, fallback to default architecture
	if arch == "" {
		arch = c.GetDefaultArchitecture()
	}

	switch arch {
	case "arm64":
		return c.GetConfig().ArchitectureConfiguration.Arm64.EmulatedMachines
	case "ppc64le":
		return c.GetConfig().ArchitectureConfiguration.Ppc64le.EmulatedMachines
	default:
		return c.GetConfig().ArchitectureConfiguration.Amd64.EmulatedMachines
	}
}
```

Reply:
Ya that's pretty close to what I had in mind.  You could also make the check a func.

Reply:
An easy enough refactor, will update. Thanks.